tools/Microsoft.AspNet.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -59,7 +59,7 @@ protected string Term(string name, int offset, int count, string array, string s
             {
                 ulong mask = 0;
                 ulong comp = 0;
-                for (var scan = 0; scan != count; ++scan)
+                for (var scan = 0; scan < count; scan++)
```

Comment:
> unconventional looping styles 

Its how you get maximum flexibility in C++ so you can swap between and array and and iterator without rewriting the loop checks; however C# has extra overheads when doing loops and uses iterators/enumerators via ```foreach```; and the jitter is happy to give up early trying to workout if a loop is bounds safe. 

There is a better breakdown somewhere from the team, which I can't find, but this may provide some insight: http://www.codeproject.com/Articles/844781/Digging-Into-NET-Loop-Performance-Bounds-checking